src/main/java/org/example/stockapitest/security/ApiKeyFilter.java

```diff
@@ -0,0 +1,57 @@
+package org.example.stockapitest.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.example.stockapitest.exception.ApiKeyException;
+import org.example.stockapitest.response.CommonResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+@Component
+public class ApiKeyFilter implements Filter {
+    private static final String API_KEY_HEADER = "x-api-key";
+    private static final String API_KEY = "c18aa07f-f005-4c2f-b6db-dff8294e6b5e";
+
+//    @Autowired
+//    private ThrottleService throttleService;
+
```

Comment:
사용하지 않는 코드는 주석처리 후 최종 PR 전 일괄 삭제 하는것이 좋습니다.
주석 코드가 한 두줄 있는게 별거 아닌 것 처럼 보여도 가독성에 상당히 방해가 됩니다.
```suggestion

//    @Autowired
//    private ThrottleService throttleService;

```